### PR TITLE
docs: add documentation to create custom metrics in plugins

### LINKED ISSUE #2294: Add documentation to create custom metrics thanks to our new `MetricsLayer`
Document the way to use `MetricsLayer` in rust plugins.

### PR DESCRIPTION
**Pull request by bnjjj**
- Fix #2294

Describe how to create custom metrics

**Checklist**

Complete the checklist (and note appropriate exceptions) before a final PR is raised.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]. It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]. Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]. Tick whichever testing boxes are applicable. If you are adding Manual Tests:
    - please document the manual testing (extensively) in the Exceptions.
    - please raise a separate issue to automate the test and label it (or ask for it to be labeled) as `manual test`

### CODE REVIEW

**Review by lennyburdette — APPROVED** [review]

**Review by Geal — APPROVED** [review]

File: NEXT_CHANGELOG.md [hunk 1]
```diff
@@ -82,8 +82,24 @@ needed anymore since there is only one useful implementation.
...
 We've changed the plugin to reduce the chances of generating memory allocations when applying regex-based header propagation rules.
 
-By [@o0Ignition0o](https://github.com/o0Ignition0o) in https://github.com/apollographql/router/pull/2389
+By [@o0Ignition0o](https://github.com/o0Ignition0o) in https://github.com/apollographql/router/pull/2392
+
+## 📚 Documentation
+
+### Add documentation to create custom metrics in plugins ([Issue #2294](https://github.com/apollographql/router/issues/2294))
+
+To create your custom metrics in [Prometheus](https://prometheus.io/) you can use the [tracing macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate an event. If you respect a specific naming for your event you'll be able to generate your own custom metrics directly in Prometheus.
```
Comment (garypen) [inline NEXT_CHANGELOG.md:95]:
```suggestion
To create your custom metrics in [Prometheus](https://prometheus.io/) you can use the [tracing macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate an event. If you observe a specific naming pattern for your event you'll be able to generate your own custom metrics directly in Prometheus.
```

File: NEXT_CHANGELOG.md [hunk 2]
```diff
@@ -82,8 +82,24 @@ needed anymore since there is only one useful implementation.
...
+
+To publish a new metric, use tracing macros to generate an event that contains following prefixes:
```
Comment (garypen) [inline NEXT_CHANGELOG.md:97]:
```suggestion
To publish a new metric, use tracing macros to generate an event that contains one of the following prefixes:
```

File: docs/source/customizations/native.mdx [hunk 3]
```diff
@@ -230,6 +230,49 @@ plugins:
     # Any values here are passed to the plugin as part of your configuration
 ```
 
+## Add custom metrics
+
+> Please make sure to [enable prometheus metrics](../configuration/metrics/#using-prometheus) in your configuration if you want to have metrics generated by the Router.
+
+### Using macros
+
+To create your custom metrics in [Prometheus](https://prometheus.io/) you can use the [tracing macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate an event. If you respect a specific naming for your event you'll be able to generate your own custom metrics directly
```
Comment (garypen) [inline docs/source/customizations/native.mdx:239]: Same suggestions as above.

**Review by garypen — COMMENTED** [review]
Looks good! Some edits for read-ability. Re-request a review if you make the changes and want me to approve.

File: docs/source/customizations/native.mdx (hunk 3 above)
Comment (bnjjj) [reply docs/source/customizations/native.mdx]: Fixed ! Thanks a lot

**Pull request merged**